### PR TITLE
Fix: ValueError Unable to create tensor

### PR DESCRIPTION
**Pull request by eryawww**
The issue lies in this single line
`self.tokenized_dataset = self.dataset.map(self.tokenize_function_normal, batched=batched, remove_columns=["text"])`
Originally, it was
`self.tokenized_dataset = self.dataset.map(self.tokenize_function_normal, batched=batched)`

By default `PPOTrainer` use `data_collator = DataCollatorWithPadding(self.processing_class)` which convert all keys as a tensor. Your code has 'text' column before, it failed during tensor conversion.

### CODE REVIEW

**Comment by eryawww** [comment]
Sorry I accidentally rebase the file 😅

**Comment by UnReAlKiNg** [comment]
Hi @eryawww ,

You suggestion works, now the `PPO_FineTune` successfully runs. I not quite sure that i understand it well, so please enlighten me if i understand it wrong.

In the source code of `Class Trainer`, it has function `def get_train_dataloader(self) -> DataLoader` that has code 
```python
if is_datasets_available() and isinstance(train_dataset, datasets.Dataset):
    train_dataset = self._remove_unused_columns(train_dataset, description="training")
else:
    data_collator = self._get_collator_with_removed_columns(data_collator, description="training")
```  
which absent in `PPO trainer`, so maybe this code can automatically remove the `["text"]` colume when training.

Moreover, maybe i do something wrong. I use the `LoRA_FineTune` to finetune the gpt2 model and save it locally, also I download the `Ray2333/gpt2-large-helpful-reward_model` as my reward model. So now my process it deploy the PPO trick on the fine-tuned model. However, the training entropy is very large, even after 10,000 iterations it still has value over 150. Do you have any idea?

**Comment by eryawww** [comment]
> In the source code of `Class Trainer`, it has function `def get_train_dataloader(self) -> DataLoader` that has code
> 
> ```python
> if is_datasets_available() and isinstance(train_dataset, datasets.Dataset):
>     train_dataset = self._remove_unused_columns(train_dataset, description="training")
> else:
>     data_collator = self._get_collator_with_removed_columns(data_collator, description="training")
> ```
> 
> which absent in `PPO trainer`, so maybe this code can automatically remove the `["text"]` colume when training.

Well, I just noticed that's actually true. A great observation! I learned something, thank you.

> Moreover, maybe i do something wrong. I use the `LoRA_FineTune` to finetune the gpt2 model and save it locally, also I download the `Ray2333/gpt2-large-helpful-reward_model` as my reward model. So now my process it deploy the PPO trick on the fine-tuned model. However, the training entropy is very large, even after 10,000 iterations it still has value over 150. Do you have any idea?

That's a really huge number. I tested openai-community/gpt2, and it only has around 6-15 nats. Does the model diverge much compared to the original model? You need to set a bigger KL penalty in that case. Otherwise, I don't really know the problem.
